metrotranscom/doorway
# feat: configure file service at app startup and separate into public …

…and private uploads.

# Pull Request Template

## Issue Overview

This PR addresses #20 


## Description

Configure the file service when the partners and public app start up. Change the configuration methods to be static. Separate the file service into two for public and private uploads with additional protections. All existing use cases use the public file service but there will be a need for a private one in the future.

## How Can This Be Tested/Reviewed?

Run existing unit tests with yarn test to verify it didn't break existing behavior. I also verified the file upload functionality still worked by bringing up the partners app and uploading a new picture for a new listing. I was also able to bring up the public site with no issues.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
